test/functionalTest/cases/1661_bad_input_on_notification_error/bad_input_on_notification_error_with_repeated_alarms.test

```diff
@@ -0,0 +1,284 @@
+
+
+echo "07. Corrupt reference in subscription (using mongo)"
+echo "==================================================="
+echo 'db.csubs.update({_id: ObjectId("'$SUB_ID'")}, { $set: { "reference": "badReferenceForSubscription" }})' | mongo $CB_DB_NAME --quiet
+echo
+echo
+
+
+echo "08. Wait six seconds to assure the subscription cache has been modified"
+echo "======================================================================="
+sleep 6
+echo
+echo
+
+
+echo "09. Update E1/A1 to provoke a notification for the invalid subscription"
+echo "======================================================================="
+payload='{
+  "A1": 12,
+  "id": "E1",
+  "type": "T1"
+}'
+orionCurl --url '/v2/entities?options=keyValues' --payload "$payload" --json
+echo
+echo
+
+
+echo "10. Check log-file to make sure a Runtime Error was provoked due to the invalid subscription"
+echo "============================================================================================"
+grep "malformed URL" /tmp/contextBroker.log | cut -d\| -f 9- | cut -d: -f 2-
+echo
+echo
+
+
+--REGEXPECT--
+01. Try to create subscription for E1/A1, but with 'reference' x.y.com:abc - see error
+======================================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 50
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "description": "Invalid URL",
+    "error": "BadRequest"
+}
+
+
+02. Try to create subscription for E1/A1, but with 'reference' without IP, just a port - see error
+==================================================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 50
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "description": "Invalid URL",
+    "error": "BadRequest"
+}
+
+
+03. Try to create subscription for E1/A1, but with 'reference' http://http://abc.a.com:1234 - see error
+=======================================================================================================
+HTTP/1.1 400 Bad Request
+Content-Length: 50
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "description": "Invalid URL",
+    "error": "BadRequest"
+}
+
+
+04. Check log-file to make sure the Bad Input alarm is active
+=============================================================
+ Raising alarm BadInput 0.0.0.0: Invalid URL
+ Repeated BadInput for 0.0.0.0: Invalid URL
+ Repeated BadInput for 0.0.0.0: Invalid URL
+
+
+05. Create a valid subscription for E1/A1
+=========================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Date: REGEX(.*)
+
+
+
+06. Check log-file to make sure the Bad Input alarm is reset
+============================================================
+ Raising alarm BadInput 0.0.0.0: Invalid URL
+ Repeated BadInput for 0.0.0.0: Invalid URL
+ Repeated BadInput for 0.0.0.0: Invalid URL
+ Releasing alarm BadInput 0.0.0.0
+
+
+07. Corrupt reference in subscription (using mongo)
```

Comment:
If you want to test only the alarm logic, probably steps 7 to 10 are not needed. However, given that they are already there :) the .test could be as it is now.

NTC